libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1009,8 +1009,10 @@ AP_GPS_UBLOX::_parse_gps(void)
             state.hdop        = _buffer.pvt.p_dop;
             state.vdop        = _buffer.pvt.p_dop;
         }
-                    
-        state.last_gps_time_ms = AP_HAL::millis();
+
+        if (_buffer.pvt.fix_type >= 2) {
```

Comment:
I think it is better to do the same as the SOL message: `if (next_fix >= AP_GPS::GPS_OK_FIX_2D)`. At this point in code the `fix_type` has been already parsed so using the parsed variable is better - specially if the parsing code above is right as `fix_type == 5` is a NO FIX state.

I'm also not sure if the line below where time week millis is set shouldn't be included inside the if too - it is in SOL when not using PVT. I'm not sure what influence setting the time week and time week millis without setting `last_gps_time_ms` has.